src/runtime/parser.c

```diff
@@ -1103,6 +1129,19 @@ bool parser_init(Parser *self) {
   return true;
 }
 
+void parser_set_language(Parser *self, const TSLanguage *language) {
```

Comment:
I see what you're saying: to remove the duplication about destroying the external scanner. So `parser_destroy` does some additional teardown that we wouldn't want to do here, but maybe we could to the reverse: `parser_destroy` could call `parser_set_language(self, NULL)`.